src/main/java/com/spring/app/services/mappers/DoubleResultMapper.java

```diff
@@ -19,16 +19,27 @@
      */
     @Override
     public Double map(List<Object> objects) throws DataNotFoundException {
-        if (objects != null && !objects.isEmpty() && objects.get(SALARY_INDEX) != null) {
-            Object object = objects.get(SALARY_INDEX);
-            if (object instanceof BigDecimal){
-                return ((BigDecimal) object).doubleValue();
-            }
-            else if (object instanceof Double){
-                return (Double) object;
-            }
-            throw new DataNotFoundException("The received value is incorrect.");
+        if (objects != null && !objects.isEmpty()) {
+            return getDouble(objects.get(SALARY_INDEX));
         }
         throw new DataNotFoundException("Data not found.");
     }
+
+    /**
+     * Resolver for an object that should contains a double value
+     *
+     * @param object - input
+     * @return converted object
+     */
+    public static Double getDouble(Object object) {
```

Comment:
seems like a utility method. Extract it as a default interface method

Reply:
I have a solution. But I do not sure if this is the best decision